api/src/play_asset_pack_request_tracker.gd

```diff
@@ -0,0 +1,70 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# The PlayAssetPackRequestTracker class generates unique signal_id integers and
+# keeps a mapping of signal_id to Request objects. This signal_id is used as an
+# identifier and passed to the Android plugin. Eventually the Android plugin will
+# emit signals along with this signal_id. In this way we can know which signal 
+# emitted from the plugin corresponds to which Request object
+#
+# ##############################################################################
+class_name PlayAssetPackRequestTracker
+extends Object
+
+const _SIGNAL_ID_MAX = 9223372036854775807
+const _SIGNAL_ID_MIN = 0
+
+var _signal_id_counter : int
+var _signal_id_to_request_map : Dictionary
+
+func _init():
+	# On first request's registration, _signal_id_counter should be incremented to _SIGNAL_ID_MIN
+	_signal_id_counter = _SIGNAL_ID_MIN - 1
+	_signal_id_to_request_map = Dictionary()
+
+func get_current_signal_id() -> int:
+	return _signal_id_counter
+
+func _increment_to_next_available_signal_id():
```

Comment:
These two functions can be simplified to one line:
```
_signal_id_counter += 1
```
However if we go with the solution above the int overflow situation is not handled. (Although that is very very unlikely, only available on some ~~games that constantly calls PlayCore methods and left running for thousands of years~~)
I'm not sure if this is over-engineered or not, but it can handle the overflow situation gracefully instead of ignoring it.

Reply:
Yeah, I don't think handling the overflow case is worth the extra complexity here. As you pointed out, this situation is absurdly unlikely since 9223372036854775807 is an absurdly high number. For reference you would have to increment this counter once a millisecond for ~290 million years to overflow this value.

Reply:
That's true. I think I have overthought this problem without contextualize. Updated this function using the increment by 1 approach.